### PR TITLE
DAT-15646       DevOps - Reusable Debian Package Creation Workflow Migration

### PR DESCRIPTION
**Pull request by jandroav**
[DAT-15646](https://datical.atlassian.net/browse/DAT-15646)

## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [x] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
 

## Description

- chore(build.yml): update Maven commands in Build & Test Java 8 and Build & Test Java non-jdk-8 jobs to exclude unnecessary modules
- chore(build.yml): remove unused upload_deb job
- chore(pom.xml): remove unused plugins and dependencies related to deb packaging
- chore(control): remove unused control file
- chore(postinst): remove unused postinst file

[DAT-15646]: https://datical.atlassian.net/browse/DAT-15646?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by jnewton03 — APPROVED** [review]

**Review by filipelautert — APPROVED** [review]

**Comment by sayaliM0412** [comment]
Let's wait before merging this as the Package artifact job is failing

**Comment by jandroav** [comment]
Please do not merge this until the Jira ticket is set to RTM

**Comment by jandroav** [comment]
@sayaliM0412 I checked the build issue here https://github.com/liquibase/liquibase/actions/runs/5822561979/job/15790327295?pr=4659 but I think it will be gone once merged to master. There are no references to `deb` stuff in this PR so I guess it is happening the same as other builds where the branch workflow is not the one that is executed in favor of the master workflow file code.

**Review by filipelautert — APPROVED** [review]

**Review by jnewton03 — APPROVED** [review]

File: .github/workflows/build.yml
Comment (jandroav) [inline .github/workflows/build.yml:473]: @sayaliM0412 this is the line that fails here https://github.com/liquibase/liquibase/actions/runs/5884307781/job/15959135849#step:9:366

**Pull request merged**
